lib/handlers/connect.js

```diff
@@ -63,10 +63,44 @@ function authenticate (arg, done) {
       client.broker.registerClient(client)
       return done()
     } else if (err) {
-      write(client, {
-        cmd: 'connack',
-        returnCode: 4
-      }, client.close.bind(client, done))
+        // This is just to maintain backward compatibilty
+        if(err && !err.returnCode){
+          write(client, {
+              cmd: 'connack',
+              returnCode: 4
+          }, client.close.bind(client, done))
+          return
+        }
+        switch(err.returnCode) {
+          case 1:
+            write(client, {
+              cmd: 'connack',
+              returnCode: 1
+            }, client.close.bind(client, done))
+            break
+
+          case 2:
+            write(client, {
+              cmd: 'connack',
+              returnCode: 2
+            }, client.close.bind(client, done))
+            break
+
+          case 3 :
+            write(client, {
+              cmd: 'connack',
+              returnCode: 3
+            }, client.close.bind(client, done))
+            break
+
+          case 4 :
+          default :
+            write(client, {
+              cmd: 'connack',
+              returnCode: 4
+            }, client.close.bind(client, done))
+            break
+      }
```

Comment:
how about a check on err.returnCode if it's between 1 and 4, and use 4 instead? We can avoid the switch.